Sources/Operators/ConcatMap.swift

```diff
@@ -0,0 +1,163 @@
+        public typealias Output = NewPublisher.Output
+        public typealias Failure = Upstream.Failure
+
+        public let upstream: Upstream
+        public let transform: Transform
+
+        public init(
+            upstream: Upstream,
+            transform: @escaping Transform
+        ) {
+            self.upstream = upstream
+            self.transform = transform
+        }
+
+        public func receive<S: Subscriber>(subscriber: S)
+        where Output == S.Input, Failure == S.Failure {
+            subscriber.receive(
+                subscription: Subscription(
+                    upstream: upstream,
+                    transform: transform,
+                    downstream: subscriber
+                )
+            )
+        }
+    }
+}
+
+// MARK: - Subscription
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+private extension Publishers.ConcatMap {
+    final class Subscription<Downstream: Subscriber>: Combine.Subscription where Downstream.Input == Output, Downstream.Failure == Failure {
+        private var sink: Sink<Downstream>?
+
+        init(
+            upstream: Upstream,
+            transform: @escaping Transform,
+            downstream: Downstream
+        ) {
+            self.sink = Sink(
+                upstream: upstream,
+                downstream: downstream,
+                transform: { transform($0) }
+            )
+        }
+
+        func request(_ demand: Subscribers.Demand) {
+            sink?.demand(demand)
+        }
+
+        func cancel() {
+            sink = nil
+        }
+    }
+}
+
+// MARK: - Sink
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+private extension Publishers.ConcatMap {
+    final class Sink<Downstream: Subscriber>: CombineExt.Sink<Upstream, Downstream> where Downstream.Input == Output, Downstream.Failure == Failure {
+        private let lock = NSRecursiveLock()
+        private let transform: Transform
+        private var activePublisher: NewPublisher?
+        private var bufferedPublishers: [NewPublisher]
+        private var cancellables: Set<AnyCancellable>
+
+        init(
+            upstream: Upstream,
+            downstream: Downstream,
+            transform: @escaping Transform
+        ) {
+            self.transform = transform
+            self.bufferedPublishers = []
+            self.cancellables = []
+            super.init(
+                upstream: upstream,
+                downstream: downstream,
+                transformFailure: { $0 }
+            )
+        }
+
+        override func receive(_ input: Upstream.Output) -> Subscribers.Demand {
+            let mapped = transform(input)
+            lock.lock()
+            defer { lock.unlock() }
+
+            if activePublisher == nil {
+                setActivePublisher(mapped)
+            } else {
+                bufferedPublishers.append(mapped)
+            }
+
+            return .unlimited
```

Comment:
Not 100% sure returning unlimited demand here is right. Any reason?

Reply:
No reason, but I'm also not 100% sure what to return here. 🤔